setup.py

```diff
@@ -270,15 +277,21 @@ def __init__(self, bazel_target):
 class BuildBazelExtension(build_ext.build_ext):
   """A command that runs Bazel to build a C/C++ extension."""
 
-  def run(self):
-    for ext in self.extensions:
-      self.bazel_build(ext)
-    command.build_ext.build_ext.run(self)  # type: ignore
+  def build_extension(self, ext: Extension) -> None:
+    """
+    This method is called by setuptools to build a single extension.
+    We override it to implement our custom Bazel build logic.
+    """
+    if not isinstance(ext, BazelExtension):
+      # If it's not our custom extension type, let setuptools handle it.
+      super().build_extension(ext)
+      return
 
-  def bazel_build(self, ext):
+    # 1. Ensure the temporary build directory exists
     if not os.path.exists(self.build_temp):
       os.makedirs(self.build_temp)
 
+    # 2. Prepare the Bazel command
```

Comment:
I am surprised to see that bazel build works here.

I have tried building with a pyproject.toml file using `python -m build` as the old path of  `python setup.py develop` is deprecated https://packaging.python.org/en/latest/discussions/setup-py-deprecated/

However to get this to work, I had to copy over a bunch of bazel files like `WORKSPACE` and `requirements_lock_3_12.txt` using a MANIFEST.in file. 

Can you help me understand that If we are not using `python -m build` for now then why do we need pyproject.toml

Reply:
AFAIK the pyproject.toml is needed to tell pip to install the dependencies needed by `setup.py` itself (such as `requests`). Without the `pyproject.toml`, `python3 setup.py develop` fails with "import request not found" because that library is not installed in the isolated build environment new in Python 3.12.

Feel free to take or ignore this PR if you intend to move to a completely new way of installation. I'm checking out vLLM and happened to found Python 3.12 editable support in PyTorch/XLA useful.

Reply:
I found out that the changes were due to update setuptools package. It removed the support for previous `python setup.py develop` command which caused the failures. I wasn't seeing this before as my setuptools package was a previous version.

Thanks for the change. Really appreciate it.